### PR TITLE
Swap gem layout

### PR DESCRIPTION
**Pull request by chris-gds**
## What?

Update `frontend` app to use `gem_layout` template from `static`.

## Why?

As part of on-going a11y work to align architecture & use Components to ensure consistency and future proof to allow a11y improvements to cascade across the site.

## Visuals

Slight changes, some manual spacing adjustments have been made to mimic current design.

### Selection of visuals:

| Dev URL      |  Before / After - Mobile  |  Desktop |
| ----------- | ----------- | ----------- |
| http://frontend.dev.gov.uk/ | Screenshot 2021-05-31 at 10 52 48       | Screenshot 2021-05-31 at 10 46 06  |
| http://frontend.dev.gov.uk/foreign-travel-advice |  Screenshot 2021-05-31 at 12 40 13  |   Screenshot 2021-05-31 at 12 40 39   |
| http://frontend.dev.gov.uk/contact-electoral-registration-office | Screenshot 2021-05-31 at 13 45 50    | Screenshot 2021-05-31 at 13 45 16       |
| http://frontend.dev.gov.uk/help/cookies |  Screenshot 2021-05-31 at 14 15 38  | Screenshot 2021-05-31 at 14 14 59  |
| http://frontend.dev.gov.uk/settle-in-the-uk |  Screenshot 2021-05-31 at 14 19 05     | Screenshot 2021-05-31 at 14 18 19  |
| http://frontend.dev.gov.uk/settle-in-the-uk/y/you-re-an-eu-eea-or-swiss-citizen |  Screenshot 2021-05-31 at 16 46 31  | Screenshot 2021-05-31 at 16 45 45  | Screenshot 2021-05-31 at 16 09 27 | http://frontend.dev.gov.uk/settle-in-the-uk/y/you-re-an-eu-eea-or-swiss-citizen |  Screenshot 2021-05-31 at 16 47 00 |   |

### Related PRs

Enhancement PR:
[Remove duplicate `<headers>`](https://github.com/alphagov/frontend/pull/2775)

Resolve e2e PR
https://github.com/alphagov/govuk_publishing_components/pull/2101

Adding additional URLs to README for ref.
https://github.com/alphagov/frontend/pull/2793

----
Converting the layout to the gem version resulted in the lookup pattern not inheriting the relevant styling correctly.  Upon inspection, this is due to elements from the gem like [`fieldset`](https://components.publishing.service.gov.uk/component-guide/fieldset) not being utilised within `frontend` and therefore not using the shared classes - this has been manually adjusted.

### Example page:
http://www.gov.uk/find-electoral-things

Screenshot 2021-05-31 at 15 06 18

### Solution

This ties into the issue raised here:
https://github.com/alphagov/govuk_publishing_components/issues/2096

Should the lookup pattern be it's own Component or use a common _combination_ of Components (`fieldset, input, button` etc) these discrepancies would reduce.

### CODE REVIEW

File: app/views/simple_smart_answers/flow.html.erb [hunk 1]
```diff
@@ -2,44 +2,47 @@
...
-  </section>
-  <% if @flow_state.completed_questions.any? %>
-    <section>
-      <%= render "govuk_publishing_components/components/heading", {
-        font_size: "m",
-        heading_level: 2,
-        margin_bottom: 4,
-        text: t("formats.simple_smart_answer.your_answers"),
-      } %>
+<main id="content" class="govuk-main-wrapper">
```
Comment (owenatgov) [inline app/views/simple_smart_answers/flow.html.erb:5]: Our [skip link component](https://components.publishing.service.gov.uk/component-guide/skip_link) looks for `#main-content`. Additionally I think `#content` has some legacy style implications.

**Review by owenatgov — COMMENTED** [review]
A few pithy comments and questions. I think quite a few of these are classes used for testing only, which is something we don't wanna do so we don't clutter our markup.

File: app/views/simple_smart_answers/flow.html.erb (hunk 1 above)
Comment (injms) [reply app/views/simple_smart_answers/flow.html.erb:5]: At the moment the [public layout component sets the skip link to link to `#content`](https://github.com/alphagov/govuk_publishing_components/blob/3b15e3c69698e2f18d95e9e59e2f998dad537779/app/views/govuk_publishing_components/components/_layout_for_public.html.erb#L66), probably for legacy reasons.

Comment (chris-gds) [reply app/views/simple_smart_answers/flow.html.erb:5]: In this instance `#content` seems apt 🤔 
- thinking we should perhaps align these on the "skip link component"?

File: app/views/shared/_base_page.html.erb [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
-  <header class="page-header govuk-grid-column-two-thirds">
-    <%= render "govuk_publishing_components/components/title", context: local_assigns[:context], title: title %>
-  </header>
-  <div class="article-container group govuk-grid-column-two-thirds">
-    <div class="content-block">
-      <div class="inner">
-        <%= yield %>
+<main id="content" role="main" class="<%= main_class if local_assigns.include?(:main_class) %> govuk-main-wrapper" <%= @lang_attribute %>>
+  <div class="govuk-grid-row">
+    <header class="page-header govuk-grid-column-two-thirds">
```
[the remark this answers is not in the document]
Comment (chris-gds) [reply app/views/shared/_base_page.html.erb]: Done ✅ (I extracted these when entering error town started)

I kept the one on the home page as this is _potentially_ debatable

File: app/views/simple_smart_answers/flow.html.erb (hunk 1 above)
Comment (owenatgov) [reply app/views/simple_smart_answers/flow.html.erb:5]: The default choice on the skip link comes from [the DS page template](https://design-system.service.gov.uk/styles/page-template/) references `#main-content` as part of their default page setup. In my opinion, we ideally want ot be following them as much as possible.

I'd strongly recommend we remove `#content` from  the layout component. That however is a big job and probably there's some big hooks to detach hidden away in our apps, so it's not a job for this PR.

Comment (chris-gds) [reply app/views/simple_smart_answers/flow.html.erb:5]: > we ideally want to be following them as much as possible.

Agreed - however I _think_ the same discrepancy is happening here too? 
https://design-system.service.gov.uk/components/skip-link/ with this being `#content` being used. 

There might be historical context for this?

Comment (owenatgov) [reply app/views/simple_smart_answers/flow.html.erb:5]: I saw that and thought about pointing it out as well, however I reckon that's a documentation choice/misstep on the design system team's part.

Comment (chris-gds) [reply app/views/simple_smart_answers/flow.html.erb:5]: >That however is a big job and probably there's some big hooks to detach hidden away in our apps, so it's not a job for this PR. 

Sure, I've [captured it here via an issue](https://github.com/alphagov/frontend/issues/2790) now.

File: test/integration/licence_test.rb [hunk 3]
```diff
@@ -69,7 +69,7 @@ class LicenceTest < ActionDispatch::IntegrationTest
         end
 
         within "#content" do
-          within ".page-header" do
```
Comment (owenatgov) [inline test/integration/licence_test.rb:72]: Always big ❤️ to removing test-only classes.

**Review by owenatgov — APPROVED** [review]
Unofficially approved 👍 I reckon we should hold off on this until Ian's point on the heading is sorted out.

File: test/integration/licence_test.rb (hunk 3 above)
Comment (chris-gds) [reply test/integration/licence_test.rb:72]: >Unofficially approved 👍 I reckon we should hold off on this until Ian's point on the heading is sorted out. 

🙌🏼  Holding tight re the header visual.

**Pull request merged**